Support semVer in wrapper task, update docs (#33567)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

### Groovy 4 TO DO

<a name="semver"></a>
### SemVer release versioning

> Starting with Gradle 9.0.0, all Gradle releases follow the [Semantic Versioning (SemVer)](https://semver.org/spec/v2.0.0.html) specification.

Author: I think it's a grey area for semVer, TBH. It boils down to the "public API" part. If we consider `@Incubating` as part of it - yes, Gradle does not follow SemVer. Otherwise, it is.

> Starting with Gradle 9.0.0, all Gradle releases follow the [Semantic Versioning (SemVer)](https://semver.org/spec/v2.0.0.html) specification.

Author: I think `@Incubating` set clear enough [expectations](https://docs.gradle.org/current/userguide/feature_lifecycle.html#sec:incubating_state).  And we say that it's not part of public API:
> Features are introduced in the incubating state to allow real-world feedback to be incorporated into the feature before making it public

> Starting with Gradle 9.0.0, versions of Gradle releases will follow the [Semantic Version](https://semver.org/spec/v2.0.0.html) specification.

Reviewer: ```suggestion
Starting with Gradle 9.0.0, all Gradle releases follow the [Semantic Versioning (SemVer)](https://semver.org/spec/v2.0.0.html) specification.
```

In short, this means that the version number will be in the form of `MAJOR.MINOR.PATCH`. 
Previously, the patch version was omitted for minor releases, and the version number was in the form of `MAJOR.MINOR`.
Note that this change will not affect the versioning of any older releases of Gradle or any future backport releases to them.

Also, Gradle version for wrapper can now be specified with a major version or a minor version only.
For example, invoking
```sh
./gradlew wrapper --gradle-version=9
```
will resolve to the latest normal Gradle version with major version 9. 

> Similarly, invoking

Reviewer: ```suggestion
```

```sh
./gradlew wrapper --gradle-version=9.1
```

> will resolve to the latest Gradle version with version 9.1.

Reviewer: ```suggestion
resolves to the latest `9.1.x` version.
```

> Older versions (e.g., `8.12`) do not follow SemVer and cannot safely interpret shorthand versions.

Author: Yeah, I had this in a [previous version](https://github.com/gradle/gradle/blob/7992fc0194af26f6a6ccbf9ec8d131f131dfce8d/platforms/documentation/docs/src/docs/release/notes.md?plain=1#L81). For me (and I guess Laura), the meaning is the same, but I'll revert the change to be more detailed for clarity.

> Older versions (e.g., `8.12`) do not follow SemVer and cannot safely interpret shorthand versions.

Reviewer: @ljacomet, my wording update was not accurate. 
@ov7a, feel free to revert.
